charts/icm-as/templates/cluster-encryption-pvc.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if eq .Values.persistence.encryption.type "cluster"  -}}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ template "icm-as.fullname" . }}-cluster-encryption-pvc
+spec:
+  accessModes:
+    - ReadOnlyMany
+  # use the storage class defined by cluster-sc.yaml
+  {{- if .Values.persistence.encryption.cluster.storageClass.create }}
+  storageClassName: "{{ template "icm-as.fullname" . }}-cluster-encryptions-sc"
+  {{- else if gt (len .Values.persistence.encryption.cluster.storageClass.existingClass) 0 }}
```

Comment:
Helms `empty` function could be used here or even let Helm handle its emptiness calculation: https://helm.sh/docs/chart_template_guide/function_list/#empty